### DISCUSSION TITLE
Frequency not a valid parameter in transient parameter sweep?

### DISCUSSION THREAD

**Discussion by oe1rsa** [ra3xdh/qucs_s#837] [Q&A] [answered]
I am trying to plot the output curves from a sweep of the frequency parameter in a transient analysis. I known in the following example I could do this better with AC analysis, but I am aiming in doing a nonlinear analysis which is only possible with transient simulation.
![grafik](https://github.com/user-attachments/assets/6b71cc45-a284-4d21-88a4-d2af813dd904)
I would have expected multiple sinusoids with different frequencies plotted one over the other. I am running qucsator, I also tried it with ngspice but it does not work, i.e. there are complaints that there is no device p1 when I try to acces it with @P1[f] and SweepModel set to true.

Is what I am trying simply not possible or am I doing something wrong? Any help appreciated!

**Comment by ra3xdh** [comment, marked as answer]
Neither Nsgpice, neither Qucsator doesn't accept such syntax for device parameter. You have to use an intermediate variable. Look at the section 5 of the quick reference guide: https://ra3xdh.github.io/pdf/qucs_s_tutorial.pdf You have to define for example `f=Freq1` for AC source and define `Param=Freq1` for simulation. Qucsator accepts only variables as parameter for parameter sweep simulation. Ngspice also suports device sweep. Model parameter sweep also could be done using intermediate variable and `@p1[f]  ` syntax is not needed anymore. Nevertheless this syntax is not valid for AC source, because the frequency is not a model parameter.

**Reply by oe1rsa** [reply]
Thank you so much! I did not really believe it would work at all, but it does.
